### PR TITLE
DRILL-6888: Move nested classes outside HashAggTemplate to allow for plain java compile option

### PR DESCRIPTION
**Pull request by Ben-Zvi**
Moved two nested classes (HashAggSpilledPartition and HashAggUpdater) from inside HashAggTemplate into independent files (using "refactor--move").

   The only additional change was to pass the allocator to the HashAggUpdater via the constructor.

### CODE REVIEW

File: exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/aggregate/HashAggUpdater.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import org.apache.drill.exec.memory.BufferAllocator;
+import org.apache.drill.exec.physical.impl.common.SpilledState;
+
+public class HashAggUpdater implements SpilledState.Updater {
+  private BufferAllocator allocator;
+  HashAggUpdater(BufferAllocator allocator) { this.allocator = allocator; }
+
+  @Override
+  public void cleanup() {
+    this.cleanup();
```
Comment (paul-rogers) [inline exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/aggregate/HashAggUpdater.java]: Infinite recursion

File: exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/aggregate/HashAggUpdater.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.physical.impl.aggregate;
+
+import org.apache.drill.exec.memory.BufferAllocator;
+import org.apache.drill.exec.physical.impl.common.SpilledState;
+
+public class HashAggUpdater implements SpilledState.Updater {
+  private BufferAllocator allocator;
```
Comment (paul-rogers) [inline exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/aggregate/HashAggUpdater.java]: private final

File: exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/aggregate/HashAggUpdater.java (hunk 1 above)
Comment (Ben-Zvi) [reply exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/aggregate/HashAggUpdater.java]: Good catch !  This is how the nested class code (in the master branch) was implemented. And then "refactor" was used to "move" this class out, so it kept this "this".
   Maybe the original intention was to call the containing class' _cleanup_, like  `HashAggTemplate.this.cleanup()`

So this method is probably never called .... 
Needs a fix anyway -- should be a no-op (no reason to call the HashAggTemplate's cleanup() ).

**Comment by Ben-Zvi** [comment]
Note that the first nested class is static, and the other is not. After extracting the first static one, the code generation still failed. I'll recreate these and add the generated code to the Jira.

**Comment by Ben-Zvi** [comment]
@paul-rogers please have a look at the Jira. Attached are two codegen files (from a simple hash agg query); one file from the master branch as is, and the other after extracting the nested static class HashAggSpilledPartition, but leaving the other non-static nested class HashAggUpdater.

File: exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/aggregate/HashAggUpdater.java (hunk 1 above)
Comment (Ben-Zvi) [reply exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/aggregate/HashAggUpdater.java]: Fixed - removed this "recursion".

**Comment by Ben-Zvi** [comment]
@paul-rogers thanks for the review; maybe all the additional changes should be done as a separate Jira that would retire the Janino, and switch to plain Java.

**Pull request closed without merging**